File: Basic/Compare two fractions.cpp
//{ Driver Code Starts

#include <iostream>
#include <regex>
using namespace std;


// } Driver Code Ends

class Solution
{
public:
    string compareFrac(string s)
    {
        // Code here
        int n = s.length();
        int passo = 0;

        string sn1 = "";
        string sd1 = "";
        string sn2 = "";
        string sd2 = "";
        
        
        for (int i=0; i<n; i++)
        {
            if (passo == 0) // num1
            {
                if (s.at(i) != '/') sn1 += s.at(i);
                else passo++;
            }
            else if (passo == 1) // den2
            {
               if (s.at(i) != ',') sd1 += s.at(i);
                else passo++;
            }
            else if (passo == 2) // num2
            {
                if (s.at(i) != '/') sn2 += s.at(i);
                else passo++;
            }
            else if (passo == 3) // den2
            {
                sd2 += s.at(i);
            }
        }
        
        int num1 = stoi(sn1);
        int den1 = stoi(sd1);
        int num2 = stoi(sn2);
        int den2 = stoi(sd2);
        
        string ans = "";

        if (((double)num1/(double)den1) > ((double)num2/(double)den2))
        {
            ans += to_string(num1);
            ans += "/";
            ans += to_string(den1);
        }
        else if (((double)num1/(double)den1) < ((double)num2/(double)den2))
        {
            ans += to_string(num2);
            ans += "/";
            ans += to_string(den2);
        }
        else
        {
            ans = "equal";
        }
        
        return ans;
        
        
        
    }
};


//{ Driver Code Starts.

int main()
{
    Solution ob;
    int t;
    cin>>t;        
    cin.ignore();
    for(int i=0;i<t;i++)
    {

        string s;
        getline(cin,s);
        
        cout << ob.compareFrac(s) << endl;



    }
    return 0;
}

// } Driver Code Ends

File: Easy/Compare two fractions.py
#User function Template for python3


class Solution:
    def compareFrac (self, s):
        # code here
        pos = str.rfind(",")
        
        first = str[ : pos]
        second = str[pos + 1 : ]
        
        #print(first, " ****** ", second)
        
        p1 = first.find("/")
        num1 = int(first[ : p1])
        den1 = int(first[p1 + 1 : ])
        
        #print(num1, den1);
        
        p2 = second.find("/")
        num2 = int(second[ : p2])
        den2 = int(second[p2 + 1 : ])
        
        #print(num2, den2)
        
        if (num1 / den1) > (num2 / den2):
            return "% s" % num1 + "/" + "% s" % den1
        elif (num1 / den1) < (num2 /den2):
            return "% s" % num2 + "/" + "% s" % den2
        else:
            return "equal"




#{ 
 # Driver Code Starts
#Initial Template for Python 3
import re

if __name__ == '__main__':
    ob = Solution()
    t = int(input())
    for _ in range(t):
        str = input()
        print(ob.compareFrac(str))

# } Driver Code Ends